toolkit/address-associations/pallet/src/lib.rs

```diff
@@ -25,6 +27,15 @@ pub struct AddressAssociationSignedMessage<PartnerChainAddress> {
 	pub genesis_utxo: UtxoId,
 }
 
+pub trait OnNewAssociation<PartnerChainAddress> {
```

Comment:
also can we provide here some default implementation for `()` that does nothing? It will allow end-user to disable it if needed